### PR TITLE
Review use of new syntax

### PR DESCRIPTION
**Pull request by AnthonySteele**
use some new syntax - `throw` expressions, `nameof` operator and an `=>` property get
refactoring guided by resharper
Should be the same only less code

### CODE REVIEW

File: src/JustEat.StatsD/DisposableTimer.cs [hunk 1]
```diff
@@ -14,17 +14,13 @@ internal class DisposableTimer : IDisposableTimer
 
         public DisposableTimer(IStatsDPublisher publisher, string statName)
         {
-            if (publisher == null)
-            {
-                throw new ArgumentNullException("publisher");
-            }
+            _publisher = publisher ?? throw new ArgumentNullException("publisher");
```
Comment (martincostello) [inline src/JustEat.StatsD/DisposableTimer.cs]: Use ```nameof(publisher)```.

File: src/JustEat.StatsD/DisposableTimer.cs [hunk 2]
```diff
@@ -14,17 +14,13 @@ internal class DisposableTimer : IDisposableTimer
...
 
             if (string.IsNullOrEmpty(statName))
             {
                 throw new ArgumentNullException("statName");
```
Comment (martincostello) [inline src/JustEat.StatsD/DisposableTimer.cs]: Might as well change this one too.

**Comment by martincostello** [comment]
LGTM.

**Comment by slang25** [comment]
Very nice

**Pull request merged**